### ISSUE TITLE
Flake in grpclb_end2end_test

### ISSUE THREAD

**Issue by bigfacebear** [grpc/grpc#19726]
Test `UpdatesWithClientLoadReportingTest.ReresolveDeadBalancer` fails. Looks like due to
```
D0723 18:59:44.913729646     531 tcp_posix.cc:1262]          cannot set inq fd=20 errno=92
D0723 18:59:44.914503159      17 tcp_posix.cc:1262]          cannot set inq fd=19 errno=92
```

```
[ RUN      ] UpdatesWithClientLoadReportingTest.ReresolveDeadBalancer
I0723 18:59:44.886767910      17 grpclb_end2end_test.cc:629] starting backend server on port 27676
I0723 18:59:44.890714084     526 server_builder.cc:309]      Synchronous server. Num CQs: 1, Min pollers: 1, Max Pollers: 2, CQ timeout (msec): 10000
I0723 18:59:44.892234851      17 grpclb_end2end_test.cc:641] backend server startup complete
I0723 18:59:44.892527089      17 grpclb_end2end_test.cc:629] starting backend server on port 27677
I0723 18:59:44.893252513     528 server_builder.cc:309]      Synchronous server. Num CQs: 1, Min pollers: 1, Max Pollers: 2, CQ timeout (msec): 10000
I0723 18:59:44.894695179      17 grpclb_end2end_test.cc:641] backend server startup complete
I0723 18:59:44.895093425      17 grpclb_end2end_test.cc:629] starting backend server on port 27678
I0723 18:59:44.895830775     530 server_builder.cc:309]      Synchronous server. Num CQs: 1, Min pollers: 1, Max Pollers: 2, CQ timeout (msec): 10000
I0723 18:59:44.897239180      17 grpclb_end2end_test.cc:641] backend server startup complete
I0723 18:59:44.897539341      17 grpclb_end2end_test.cc:629] starting backend server on port 27679
I0723 18:59:44.898355639     532 server_builder.cc:309]      Synchronous server. Num CQs: 1, Min pollers: 1, Max Pollers: 2, CQ timeout (msec): 10000
I0723 18:59:44.899771750      17 grpclb_end2end_test.cc:641] backend server startup complete
I0723 18:59:44.900044551      17 grpclb_end2end_test.cc:629] starting balancer server on port 27680
I0723 18:59:44.900769578     534 server_builder.cc:309]      Synchronous server. Num CQs: 1, Min pollers: 1, Max Pollers: 2, CQ timeout (msec): 10000
I0723 18:59:44.902131525      17 grpclb_end2end_test.cc:641] balancer server startup complete
I0723 18:59:44.902417240      17 grpclb_end2end_test.cc:629] starting balancer server on port 27681
I0723 18:59:44.903155064     536 server_builder.cc:309]      Synchronous server. Num CQs: 1, Min pollers: 1, Max Pollers: 2, CQ timeout (msec): 10000
I0723 18:59:44.904479687      17 grpclb_end2end_test.cc:641] balancer server startup complete
I0723 18:59:44.904752398      17 grpclb_end2end_test.cc:629] starting balancer server on port 27682
I0723 18:59:44.907626553     538 server_builder.cc:309]      Synchronous server. Num CQs: 1, Min pollers: 1, Max Pollers: 2, CQ timeout (msec): 10000
I0723 18:59:44.908930164      17 grpclb_end2end_test.cc:641] balancer server startup complete
D0723 18:59:44.913729646     531 tcp_posix.cc:1262]          cannot set inq fd=20 errno=92
D0723 18:59:44.914503159      17 tcp_posix.cc:1262]          cannot set inq fd=19 errno=92
I0723 18:59:44.917820798      17 subchannel.cc:1082]         New connected subchannel at 0x7b0c00008040 for subchannel 0x7b500003aa00
I0723 18:59:44.921765206     535 grpclb_end2end_test.cc:212] LB[0x7b4c00001c10]: BalanceLoad
I0723 18:59:44.922439992     535 grpclb_end2end_test.cc:228] LB[0x7b4c00001c10]: received initial message 'initial_request {
  name: "application_target_name"
}
'
I0723 18:59:44.923467734      17 grpclb_end2end_test.cc:1691] ========= BEFORE FIRST BATCH ==========
test/cpp/end2end/grpclb_end2end_test.cc:609: Failure
Value of: status.ok()
  Actual: false
Expected: true
code=4 message=Deadline Exceeded
test/cpp/end2end/grpclb_end2end_test.cc:611: Failure
Expected equality of these values:
  response.message()
    Which is: ""
  kRequestMessage_
    Which is: "Live long and prosper."
I0723 18:59:54.910725796     525 grpclb.cc:1590]             [grpclb 0x7b5800061200] No response from balancer after fallback timeout; entering fallback mode
test/cpp/end2end/grpclb_end2end_test.cc:609: Failure
Value of: status.ok()
  Actual: false
Expected: true
code=14 message=Empty update
test/cpp/end2end/grpclb_end2end_test.cc:611: Failure
Expected equality of these values:
  response.message()
    Which is: ""
  kRequestMessage_
    Which is: "Live long and prosper."
test/cpp/end2end/grpclb_end2end_test.cc:609: Failure
Value of: status.ok()
  Actual: false
Expected: true
code=14 message=Empty update
test/cpp/end2end/grpclb_end2end_test.cc:611: Failure
Expected equality of these values:
  response.message()
    Which is: ""
  kRequestMessage_
    Which is: "Live long and prosper."
test/cpp/end2end/grpclb_end2end_test.cc:609: Failure
Value of: status.ok()
  Actual: false
Expected: true
code=14 message=Empty update
test/cpp/end2end/grpclb_end2end_test.cc:611: Failure
Expected equality of these values:
  response.message()
    Which is: ""
  kRequestMessage_
    Which is: "Live long and prosper."
test/cpp/end2end/grpclb_end2end_test.cc:609: Failure
Value of: status.ok()
  Actual: false
Expected: true
code=14 message=Empty update
test/cpp/end2end/grpclb_end2end_test.cc:611: Failure
Expected equality of these values:
  response.message()
    Which is: ""
  kRequestMessage_
    Which is: "Live long and prosper."
test/cpp/end2end/grpclb_end2end_test.cc:609: Failure
Value of: status.ok()
  Actual: false
Expected: true
code=14 message=Empty update
test/cpp/end2end/grpclb_end2end_test.cc:611: Failure
Expected equality of these values:
  response.message()
    Which is: ""
  kRequestMessage_
    Which is: "Live long and prosper."
test/cpp/end2end/grpclb_end2end_test.cc:609: Failure
Value of: status.ok()
  Actual: false
Expected: true
code=14 message=Empty update
test/cpp/end2end/grpclb_end2end_test.cc:611: Failure
Expected equality of these values:
  response.message()
    Which is: ""
  kRequestMessage_
    Which is: "Live long and prosper."
test/cpp/end2end/grpclb_end2end_test.cc:609: Failure
Value of: status.ok()
  Actual: false
Expected: true
code=14 message=Empty update
test/cpp/end2end/grpclb_end2end_test.cc:611: Failure
Expected equality of these values:
  response.message()
    Which is: ""
  kRequestMessage_
    Which is: "Live long and prosper."
test/cpp/end2end/grpclb_end2end_test.cc:609: Failure
Value of: status.ok()
  Actual: false
Expected: true
code=14 message=Empty update
test/cpp/end2end/grpclb_end2end_test.cc:611: Failure
Expected equality of these values:
  response.message()
    Which is: ""
  kRequestMessage_
    Which is: "Live long and prosper."
test/cpp/end2end/grpclb_end2end_test.cc:609: Failure
Value of: status.ok()
  Actual: false
Expected: true
code=14 message=Empty update
test/cpp/end2end/grpclb_end2end_test.cc:611: Failure
Expected equality of these values:
  response.message()
    Which is: ""
  kRequestMessage_
    Which is: "Live long and prosper."
I0723 18:59:54.918900244      17 grpclb_end2end_test.cc:1693] ========= DONE WITH FIRST BATCH ==========
test/cpp/end2end/grpclb_end2end_test.cc:1695: Failure
Expected equality of these values:
  10U
    Which is: 10
  backends_[0]->service_.request_count()
    Which is: 0
I0723 18:59:54.919174578      17 grpclb_end2end_test.cc:1698] ********** ABOUT TO KILL BACKEND 0 *************
I0723 18:59:54.919231532      17 grpclb_end2end_test.cc:662] backend about to shutdown
I0723 18:59:54.922517322      17 grpclb_end2end_test.cc:666] backend shutdown completed
I0723 18:59:54.922546445      17 grpclb_end2end_test.cc:1700] ********** KILLED BACKEND 0 *************
test/cpp/end2end/grpclb_end2end_test.cc:1707: Failure
Expected equality of these values:
  1U
    Which is: 1
  balancers_[0]->service_.response_count()
    Which is: 0
I0723 18:59:54.923366163      17 grpclb_end2end_test.cc:1714] ********** ABOUT TO KILL BALANCER 0 *************
I0723 18:59:54.923392191      17 grpclb_end2end_test.cc:662] balancer about to shutdown
I0723 18:59:54.923597481      17 grpclb_end2end_test.cc:302] LB[0x7b4c00001c10]: shut down
I0723 18:59:54.926853668      17 chttp2_transport.cc:1805]   ipv4:127.0.0.1:60724: Sending goaway err={"created":"@1563908394.926491939","description":"Server shutdown","file":"src/core/lib/surface/server.cc","file_line":321,"grpc_status":0}
I0723 18:59:54.927768190     529 chttp2_transport.cc:1161]   ipv4:127.0.0.1:27680: Got goaway [0] err={"created":"@1563908394.927582699","description":"GOAWAY received","file":"src/core/ext/transport/chttp2/transport/chttp2_transport.cc","file_line":1152,"grpc_status":14,"http2_error":0,"raw_bytes":"Server shutdown"}
I0723 18:59:54.929475516     535 grpclb_end2end_test.cc:283] LB[0x7b4c00001c10]: done
I0723 18:59:54.934519421      17 grpclb_end2end_test.cc:666] balancer shutdown completed
I0723 18:59:54.934545989      17 grpclb_end2end_test.cc:1716] ********** KILLED BALANCER 0 *************
D0723 18:59:54.935122072     529 tcp_posix.cc:1262]          cannot set inq fd=6 errno=92
D0723 18:59:54.935558122     533 tcp_posix.cc:1262]          cannot set inq fd=5 errno=92
I0723 18:59:54.937085921     531 subchannel.cc:1082]         New connected subchannel at 0x7b0c00006b40 for subchannel 0x7b500003c200
I0723 18:59:54.948427606     537 grpclb_end2end_test.cc:212] LB[0x7b4c000c5190]: BalanceLoad
I0723 18:59:54.949284708     537 grpclb_end2end_test.cc:228] LB[0x7b4c000c5190]: received initial message 'initial_request {
  name: "application_target_name"
}
'
I0723 18:59:54.950846426     537 grpclb_end2end_test.cc:347] LB[0x7b4c000c5190]: sleeping for 0 ms...
I0723 18:59:54.951402664     537 grpclb_end2end_test.cc:351] LB[0x7b4c000c5190]: Woke up! Sending response 'server_list {
  servers {
    ip_address: "\177\000\000\001"
    port: 27677
    load_balance_token: "token063"
  }
}
'
I0723 18:59:54.952286476     529 grpclb.cc:1084]             [grpclb 0x7b5800061200] Received response from balancer; exiting fallback mode
D0723 18:59:54.953984991     533 tcp_posix.cc:1262]          cannot set inq fd=13 errno=92
D0723 18:59:54.954365047     539 tcp_posix.cc:1262]          cannot set inq fd=14 errno=92
I0723 18:59:54.957039566     529 subchannel.cc:1082]         New connected subchannel at 0x7b0c000141f0 for subchannel 0x7b500003b400
I0723 18:59:54.961712892      17 grpclb_end2end_test.cc:1723] ========= BEFORE SECOND BATCH ==========
I0723 18:59:54.988130053      17 grpclb_end2end_test.cc:1725] ========= DONE WITH SECOND BATCH ==========
test/cpp/end2end/grpclb_end2end_test.cc:1730: Failure
Expected equality of these values:
  1U
    Which is: 1
  balancers_[0]->service_.response_count()
    Which is: 0
I0723 18:59:54.988564363      17 grpclb_end2end_test.cc:662] backend about to shutdown
I0723 18:59:54.989361745      17 chttp2_transport.cc:1805]   ipv4:127.0.0.1:51258: Sending goaway err={"created":"@1563908394.989133973","description":"Server shutdown","file":"src/core/lib/surface/server.cc","file_line":321,"grpc_status":0}
I0723 18:59:54.991030942     539 chttp2_transport.cc:1161]   ipv4:127.0.0.1:27677: Got goaway [0] err={"created":"@1563908394.990871567","description":"GOAWAY received","file":"src/core/ext/transport/chttp2/transport/chttp2_transport.cc","file_line":1152,"grpc_status":14,"http2_error":0,"raw_bytes":"Server shutdown"}
I0723 18:59:54.992260100     539 subchannel.cc:1031]         Connect failed: {"created":"@1563908394.991954206","description":"Failed to connect to remote host: Connection refused","errno":111,"file":"src/core/lib/iomgr/tcp_client_posix.cc","file_line":207,"os_error":"Connection refused","syscall":"connect","target_address":"ipv4:127.0.0.1:27677"}
I0723 18:59:54.992588088     539 subchannel.cc:970]          Subchannel 0x7b500003b400: Retry in 1000 milliseconds
I0723 18:59:54.994535581      17 grpclb_end2end_test.cc:666] backend shutdown completed
I0723 18:59:54.994561686      17 grpclb_end2end_test.cc:662] backend about to shutdown
I0723 18:59:54.995884992      17 grpclb_end2end_test.cc:666] backend shutdown completed
I0723 18:59:54.995911460      17 grpclb_end2end_test.cc:662] backend about to shutdown
I0723 18:59:54.997295211      17 grpclb_end2end_test.cc:666] backend shutdown completed
I0723 18:59:54.997325666      17 grpclb_end2end_test.cc:662] balancer about to shutdown
I0723 18:59:54.997374037      17 grpclb_end2end_test.cc:302] LB[0x7b4c000c5190]: shut down
I0723 18:59:54.997924991      17 chttp2_transport.cc:1805]   ipv4:127.0.0.1:51228: Sending goaway err={"created":"@1563908394.997783913","description":"Server shutdown","file":"src/core/lib/surface/server.cc","file_line":321,"grpc_status":0}
I0723 18:59:54.998777987     537 grpclb_end2end_test.cc:283] LB[0x7b4c000c5190]: done
I0723 18:59:54.999094377     539 chttp2_transport.cc:1161]   ipv4:127.0.0.1:27681: Got goaway [0] err={"created":"@1563908394.998984610","description":"GOAWAY received","file":"src/core/ext/transport/chttp2/transport/chttp2_transport.cc","file_line":1152,"grpc_status":14,"http2_error":0,"raw_bytes":"Server shutdown"}
I0723 18:59:55.003562542      17 grpclb_end2end_test.cc:666] balancer shutdown completed
I0723 18:59:55.003879149     539 subchannel.cc:1031]         Connect failed: {"created":"@1563908395.003581659","description":"Failed to connect to remote host: Connection refused","errno":111,"file":"src/core/lib/iomgr/tcp_client_posix.cc","file_line":207,"os_error":"Connection refused","syscall":"connect","target_address":"ipv4:127.0.0.1:27681"}
I0723 18:59:55.003892046      17 grpclb_end2end_test.cc:662] balancer about to shutdown
I0723 18:59:55.004116457      17 grpclb_end2end_test.cc:302] LB[0x7b4c00011f10]: shut down
I0723 18:59:55.004327069     539 subchannel.cc:970]          Subchannel 0x7b500002a200: Retry in 997 milliseconds
I0723 18:59:55.006913159      17 grpclb_end2end_test.cc:666] balancer shutdown completed
[  FAILED  ] UpdatesWithClientLoadReportingTest.ReresolveDeadBalancer (10123 ms)
```
Log link: https://source.cloud.google.com/results/invocations/d28dec7b-d6da-493b-adef-dd4a84bdfc27/targets

**Comment by bigfacebear** [comment]
Also seen some other tests in `grpclb_end2end_test` flake.

`SingleBalancerTest.SecureNamingDeathTest`: https://source.cloud.google.com/results/invocations/20f78ca3-ea7c-42c2-99f4-4c419b69bbf0/targets

`SingleBalancerTest.BackendsRestart` flakes on the local machine. Log message:
```
D0724 11:49:02.915233717    7526 test_config.cc:384]         test slowdown factor: sanitizer=1, fixture=1, poller=1, total=1
Note: Google Test filter = SingleBalancerTest.BackendsRestart
[==========] Running 1 test from 1 test case.
[----------] Global test environment set-up.
[----------] 1 test from SingleBalancerTest
D0724 11:49:02.915601605    7526 ev_posix.cc:174]            Using polling engine: poll
D0724 11:49:02.915647380    7526 dns_resolver_ares.cc:485]   Using ares dns resolver
[ RUN      ] SingleBalancerTest.BackendsRestart
I0724 11:49:02.918743673    7526 grpclb_end2end_test.cc:629] starting backend server on port 29854
I0724 11:49:02.919006133    7551 server_builder.cc:309]      Synchronous server. Num CQs: 1, Min pollers: 1, Max Pollers: 2, CQ timeout (msec): 10000
I0724 11:49:02.920028301    7526 grpclb_end2end_test.cc:641] backend server startup complete
I0724 11:49:02.924291094    7526 grpclb_end2end_test.cc:629] starting backend server on port 2429
I0724 11:49:02.924761889    7569 server_builder.cc:309]      Synchronous server. Num CQs: 1, Min pollers: 1, Max Pollers: 2, CQ timeout (msec): 10000
I0724 11:49:02.925727991    7526 grpclb_end2end_test.cc:641] backend server startup complete
I0724 11:49:02.930872471    7526 grpclb_end2end_test.cc:629] starting backend server on port 24424
I0724 11:49:02.931809449    7599 server_builder.cc:309]      Synchronous server. Num CQs: 1, Min pollers: 1, Max Pollers: 2, CQ timeout (msec): 10000
I0724 11:49:02.933215442    7526 grpclb_end2end_test.cc:641] backend server startup complete
I0724 11:49:02.936461938    7526 grpclb_end2end_test.cc:629] starting backend server on port 29354
I0724 11:49:02.936647223    7614 server_builder.cc:309]      Synchronous server. Num CQs: 1, Min pollers: 1, Max Pollers: 2, CQ timeout (msec): 10000
I0724 11:49:02.936898185    7526 grpclb_end2end_test.cc:641] backend server startup complete
I0724 11:49:02.937893883    7526 grpclb_end2end_test.cc:629] starting balancer server on port 10757
I0724 11:49:02.938285852    7619 server_builder.cc:309]      Synchronous server. Num CQs: 1, Min pollers: 1, Max Pollers: 2, CQ timeout (msec): 10000
I0724 11:49:02.938841079    7526 grpclb_end2end_test.cc:641] balancer server startup complete
I0724 11:49:02.939782004    7526 subchannel.cc:1082]         New connected subchannel at 0x561d8b3eab10 for subchannel 0x561d8b3e79e0
I0724 11:49:02.940113395    7621 grpclb_end2end_test.cc:212] LB[0x561d8b3bef68]: BalanceLoad
I0724 11:49:02.940272149    7621 grpclb_end2end_test.cc:228] LB[0x561d8b3bef68]: received initial message 'initial_request {
  name: "application_target_name"
}
'
I0724 11:49:02.940284615    7621 grpclb_end2end_test.cc:347] LB[0x561d8b3bef68]: sleeping for 0 ms...
I0724 11:49:02.940308802    7621 grpclb_end2end_test.cc:351] LB[0x561d8b3bef68]: Woke up! Sending response 'server_list {
  servers {
    ip_address: "\177\000\000\001"
    port: 29854
    load_balance_token: "token001"
  }
  servers {
    ip_address: "\177\000\000\001"
    port: 2429
    load_balance_token: "token002"
  }
  servers {
    ip_address: "\177\000\000\001"
    port: 24424
    load_balance_token: "token003"
  }
  servers {
    ip_address: "\177\000\000\001"
    port: 29354
    load_balance_token: "token004"
  }
}
'
0724 11:49:02.940917481    7526 subchannel.cc:1082]         New connected subchannel at 0x561d8b3fec60 for subchannel 0x561d8b3ea590
I0724 11:49:02.941061643    7526 subchannel.cc:1082]         New connected subchannel at 0x561d8b4007f0 for subchannel 0x561d8b3f89e0
I0724 11:49:02.941081742    7526 subchannel.cc:1082]         New connected subchannel at 0x561d8b400aa0 for subchannel 0x561d8b3fa460
I0724 11:49:02.941385243    7526 subchannel.cc:1082]         New connected subchannel at 0x561d8b402360 for subchannel 0x561d8b3fb4b0
I0724 11:49:03.010061730    7526 grpclb_end2end_test.cc:662] backend about to shutdown
I0724 11:49:03.010244447    7526 chttp2_transport.cc:1805]   ipv4:127.0.0.1:50486: Sending goaway err={"created":"@1563994143.010222508","description":"Server shutdown","file":"src/core/lib/surface/server.cc","file_line":321,"grpc_status":0}
I0724 11:49:03.010427842    7526 grpclb_end2end_test.cc:666] backend shutdown completed
I0724 11:49:03.010436574    7526 grpclb_end2end_test.cc:662] backend about to shutdown
I0724 11:49:03.010581686    7526 chttp2_transport.cc:1805]   ipv4:127.0.0.1:46680: Sending goaway err={"created":"@1563994143.010565967","description":"Server shutdown","file":"src/core/lib/surface/server.cc","file_line":321,"grpc_status":0}
I0724 11:49:03.010660667    7542 chttp2_transport.cc:1161]   ipv4:127.0.0.1:29854: Got goaway [0] err={"created":"@1563994143.010644205","description":"GOAWAY received","file":"src/core/ext/transport/chttp2/transport/chttp2_transport.cc","file_line":1152,"grpc_status":14,"http2_error":0,"raw_bytes":"Server shutdown"}
I0724 11:49:03.010699181    7542 chttp2_transport.cc:1161]   ipv4:127.0.0.1:2429: Got goaway [0] err={"created":"@1563994143.010690399","description":"GOAWAY received","file":"src/core/ext/transport/chttp2/transport/chttp2_transport.cc","file_line":1152,"grpc_status":14,"http2_error":0,"raw_bytes":"Server shutdown"}
I0724 11:49:03.010725909    7526 grpclb_end2end_test.cc:666] backend shutdown completed
I0724 11:49:03.010733670    7526 grpclb_end2end_test.cc:662] backend about to shutdown
I0724 11:49:03.010873743    7526 chttp2_transport.cc:1805]   ipv4:127.0.0.1:42172: Sending goaway err={"created":"@1563994143.010859899","description":"Server shutdown","file":"src/core/lib/surface/server.cc","file_line":321,"grpc_status":0}
I0724 11:49:03.011100522    7526 grpclb_end2end_test.cc:666] backend shutdown completed
I0724 11:49:03.011109990    7526 grpclb_end2end_test.cc:662] backend about to shutdown
I0724 11:49:03.011587074    7526 chttp2_transport.cc:1805]   ipv4:127.0.0.1:36434: Sending goaway err={"created":"@1563994143.011254277","description":"Server shutdown","file":"src/core/lib/surface/server.cc","file_line":321,"grpc_status":0}
I0724 11:49:03.011826855    7526 grpclb_end2end_test.cc:666] backend shutdown completed
I0724 11:49:03.011936875    7526 chttp2_transport.cc:1161]   ipv4:127.0.0.1:29354: Got goaway [0] err={"created":"@1563994143.011927573","description":"GOAWAY received","file":"src/core/ext/transport/chttp2/transport/chttp2_transport.cc","file_line":1152,"grpc_status":14,"http2_error":0,"raw_bytes":"Server shutdown"}
I0724 11:49:03.012071050    7526 grpclb_end2end_test.cc:629] starting backend server on port 29854
I0724 11:49:03.012135512    7791 server_builder.cc:309]      Synchronous server. Num CQs: 1, Min pollers: 1, Max Pollers: 2, CQ timeout (msec): 10000
I0724 11:49:03.012524723    7526 grpclb_end2end_test.cc:641] backend server startup complete
I0724 11:49:03.012538843    7526 grpclb_end2end_test.cc:629] starting backend server on port 2429
I0724 11:49:03.012849705    7793 server_builder.cc:309]      Synchronous server. Num CQs: 1, Min pollers: 1, Max Pollers: 2, CQ timeout (msec): 10000
I0724 11:49:03.013131100    7526 grpclb_end2end_test.cc:641] backend server startup complete
I0724 11:49:03.013144121    7526 grpclb_end2end_test.cc:629] starting backend server on port 24424
I0724 11:49:03.013243283    7796 server_builder.cc:309]      Synchronous server. Num CQs: 1, Min pollers: 1, Max Pollers: 2, CQ timeout (msec): 10000
I0724 11:49:03.013488405    7526 grpclb_end2end_test.cc:641] backend server startup complete
I0724 11:49:03.013498653    7526 grpclb_end2end_test.cc:629] starting backend server on port 29354
I0724 11:49:03.013667721    7798 server_builder.cc:309]      Synchronous server. Num CQs: 1, Min pollers: 1, Max Pollers: 2, CQ timeout (msec): 10000
I0724 11:49:03.014156993    7526 grpclb_end2end_test.cc:641] backend server startup complete
I0724 11:49:03.014248543    7526 chttp2_transport.cc:1161]   ipv4:127.0.0.1:24424: Got goaway [0] err={"created":"@1563994143.014237401","description":"GOAWAY received","file":"src/core/ext/transport/chttp2/transport/chttp2_transport.cc","file_line":1152,"grpc_status":14,"http2_error":0,"raw_bytes":"Server shutdown"}
test/cpp/end2end/grpclb_end2end_test.cc:609: Failure
Value of: status.ok()
  Actual: false
Expected: true
code=14 message=Socket closed
test/cpp/end2end/grpclb_end2end_test.cc:611: Failure
Expected equality of these values:
  response.message()
    Which is: ""
  kRequestMessage_
    Which is: "Live long and prosper."
I0724 11:49:03.014549269    7526 grpclb_end2end_test.cc:662] backend about to shutdown
I0724 11:49:03.014683191    7542 subchannel.cc:1031]         Connect failed: {"created":"@1563994143.014618213","description":"Failed to connect to remote host: Connection refused","errno":111,"file":"src/core/lib/iomgr/tcp_client_posix.cc","file_line":207,"os_error":"Connection refused","syscall":"connect","target_address":"ipv4:127.0.0.1:29854"}
I0724 11:49:03.014704807    7542 subchannel.cc:1031]         Connect failed: {"created":"@1563994143.014632214","description":"Failed to connect to remote host: Connection refused","errno":111,"file":"src/core/lib/iomgr/tcp_client_posix.cc","file_line":207,"os_error":"Connection refused","syscall":"connect","target_address":"ipv4:127.0.0.1:2429"}
I0724 11:49:03.014719253    7542 subchannel.cc:1031]         Connect failed: {"created":"@1563994143.014639594","description":"Failed to connect to remote host: Connection refused","errno":111,"file":"src/core/lib/iomgr/tcp_client_posix.cc","file_line":207,"os_error":"Connection refused","syscall":"connect","target_address":"ipv4:127.0.0.1:29354"}
I0724 11:49:03.014755765    7542 subchannel.cc:970]          Subchannel 0x561d8b3ea590: Retry in 996 milliseconds
I0724 11:49:03.014762794    7542 subchannel.cc:970]          Subchannel 0x561d8b3f89e0: Retry in 996 milliseconds
I0724 11:49:03.014768443    7542 subchannel.cc:970]          Subchannel 0x561d8b3fb4b0: Retry in 997 milliseconds
I0724 11:49:03.014765525    7526 grpclb_end2end_test.cc:666] backend shutdown completed
I0724 11:49:03.014786578    7526 grpclb_end2end_test.cc:662] backend about to shutdown
I0724 11:49:03.014865058    7526 grpclb_end2end_test.cc:666] backend shutdown completed
I0724 11:49:03.014876051    7526 grpclb_end2end_test.cc:662] backend about to shutdown
D0724 11:49:03.015646922    7797 chttp2_server.cc:122]       Handshaking failed: "No Error"
I0724 11:49:03.016677546    7542 subchannel.cc:1082]         New connected subchannel at 0x7f21cc002020 for subchannel 0x561d8b3fa460
I0724 11:49:03.017761009    7542 subchannel.cc:1031]         Connect failed: {"created":"@1563994143.017736164","description":"Failed to connect to remote host: Connection refused","errno":111,"file":"src/core/lib/iomgr/tcp_client_posix.cc","file_line":207,"os_error":"Connection refused","syscall":"connect","target_address":"ipv4:127.0.0.1:24424"}
I0724 11:49:03.017780413    7542 subchannel.cc:970]          Subchannel 0x561d8b3fa460: Retry in 999 milliseconds
I0724 11:49:03.022651828    7526 grpclb_end2end_test.cc:666] backend shutdown completed
I0724 11:49:03.022672633    7526 grpclb_end2end_test.cc:662] backend about to shutdown
I0724 11:49:03.024220639    7526 grpclb_end2end_test.cc:666] backend shutdown completed
I0724 11:49:03.024240193    7526 grpclb_end2end_test.cc:662] balancer about to shutdown
I0724 11:49:03.024249341    7526 grpclb_end2end_test.cc:302] LB[0x561d8b3bef68]: shut down
I0724 11:49:03.024260333    7621 grpclb_end2end_test.cc:283] LB[0x561d8b3bef68]: done
I0724 11:49:03.024287902    7526 chttp2_transport.cc:1805]   ipv4:127.0.0.1:59482: Sending goaway err={"created":"@1563994143.024273039","description":"Server shutdown","file":"src/core/lib/surface/server.cc","file_line":321,"grpc_status":0}
I0724 11:49:03.024629909    7542 chttp2_transport.cc:1161]   ipv4:127.0.0.1:10757: Got goaway [0] err={"created":"@1563994143.024618113","description":"GOAWAY received","file":"src/core/ext/transport/chttp2/transport/chttp2_transport.cc","file_line":1152,"grpc_status":14,"http2_error":0,"raw_bytes":"Server shutdown"}
I0724 11:49:03.033013690    7526 grpclb_end2end_test.cc:666] balancer shutdown completed
[  FAILED  ] SingleBalancerTest.BackendsRestart (118 ms)
[----------] 1 test from SingleBalancerTest (118 ms total)

[----------] Global test environment tear-down
[==========] 1 test from 1 test case ran. (118 ms total)
[  PASSED  ] 0 tests.
[  FAILED  ] 1 test, listed below:
[  FAILED  ] SingleBalancerTest.BackendsRestart

 1 FAILED TEST
D0724 11:49:03.042702046    7526 ev_posix.cc:174]            Using polling engine: poll
D0724 11:49:03.042732258    7526 dns_resolver_ares.cc:485]   Using ares dns resolver
FAILED: bins/opt/grpclb_end2end_test --gtest_filter=SingleBalancerTest.BackendsRestart  GRPC_POLL_STRATEGY=poll [ret=1, pid=7526, time=0.1sec]
```

**Comment by AspirinSJL** [comment]
I don't see obvious reason of this flake. I will come back to this bug if it becomes more flaky.

**Comment by lidizheng** [comment]
Observing this flake in https://github.com/grpc/grpc/pull/21819
https://source.cloud.google.com/results/invocations/323d49d2-d55a-428e-ab20-fa2d600ca4a2/targets/%2F%2Ftest%2Fcpp%2Fend2end:grpclb_end2end_test@poller%3Depoll1/tests;group=UpdatesWithClientLoadReportingTest;test=ReresolveDeadBalancer;row=1

**Comment by markdroth** [comment]
This was fixed in #22535.
